HIP/hip-0000-introduction-of-blocks.md

```diff
@@ -0,0 +1,199 @@
+# Introduction of Blocks
+
+---
+hip: 0000 
+title: Introduction of Blocks
+author: Daniel Ivanov <daniel-k-ivanov95@gmail.tech>, Ivan Kavaldzhiev <ivan.kavaldzhiev@limechain.tech>
+type: Standards Track 
+category: Core 
+needs-council-approval: Yes 
+status: Draft
+created: 2022-04-03 
+discussions-to: //TODO 
+updated: 
+---
+
+## Abstract
+
+Specifies how to introduce and formalize the concepts of Blocks in Hedera Hashgraph so that it can be used as a foundation on which further interoperability with existing DLT networks and infrastructure can be built.
+
+## Motivation
+
+The concept of blocks is a vital part of the existing Ethereum infrastructure as such, the introduction of blocks in Hedera can be considered a foundational step towards greater interoperability with EVM based tooling, explorers, exchanges and wallet providers.
+
+## Rationale
+
+Hedera must have a single consistent answer to what a block number is, the hash and which transactions are included in that block number. This is required for two reasons:
+
+1. The Smart Contracts Service to have the context of a `number` and the `hash` of the current block while running the EVM bytecode.
+2. Mirror Nodes to have the information in order to implement standard JSON RPC endpoints and be able to answer queries such as:
+    1. Getting current block number
+    2. Getting block by hash/number
+    3. Getting a list of transactions for a block by number/hash
+    4. Getting logs based on block number filtering
+
+**Design Goal #1 Minimize Changes**
+
+The Block concept should fit naturally into the existing processes, mechanisms and state updates. It must keep the same responsibilities between consensus, services and mirror nodes.
+
+**Design Goal #2 Lightweight**
+
+The Block concept must not add a lot of complexity and performance overhead to the processing of transactions. It must have minimal impact on the TPS of the network.
+
+Based on the described design goals above, the outlined specification defines that block properties are to be computed and populated at different points in time and by different components of the network, based on their responsibility.
+
+## Specification
+
+**Definitions**
+
+- `block` → `Record file` containing all `Record Stream Objects` grouped in a 2-second time frame.
+- `block number` → consecutive number of the `Record file` that is being incremented by `1` for every new `Record file`. This value can be initially bootstrapped through `Mirror Nodes` and after that maintained by services nodes.
+- `block hash` → The `32 byte keccak256 running hash` of the `Record file`. That is the running hash of the last `Record Stream Object` from the previous `Record file`.
```

Comment:
Platform currently supports only a SHA-384 (SHA-2) running hash; switching to keccak256 would have significant ripple effects for mirror nodes I think.

Reply:
The HIP has been updated to use a 32 byte prefix of the running hash